teku/src/main/java/tech/pegasys/teku/Teku.java

```diff
@@ -19,23 +19,38 @@
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
 import tech.pegasys.teku.bls.impl.blst.BlstLoader;
 import tech.pegasys.teku.cli.BeaconNodeCommand;
+import tech.pegasys.teku.cli.BeaconNodeCommand.StartAction;
 import tech.pegasys.teku.config.TekuConfiguration;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
 
-public final class Teku {
+public final class Teku implements TekuFacade {
 
-  public static void main(final String... args) {
-    Thread.setDefaultUncaughtExceptionHandler(new TekuDefaultExceptionHandler());
-    Security.addProvider(new BouncyCastleProvider());
-    final PrintWriter outputWriter = new PrintWriter(System.out, true, Charset.defaultCharset());
-    final PrintWriter errorWriter = new PrintWriter(System.err, true, Charset.defaultCharset());
-    final int result =
-        new BeaconNodeCommand(outputWriter, errorWriter, System.getenv(), Teku::start).parse(args);
+  static final Teku INSTANCE = new Teku();
```

Comment:
Actually we can create multiple nodes via `Teku.start*()`. The original idea was to have `Teku` singleton in the role of 'node factory'. 
I'll try to think about better design though

Reply:
Ah I see. I'm ok with the way it is not but another option might be to just make the methods on `Teku` static instead?

I think then if possible I'd just have the `main` method add the shutdown hook and leave it to the embedding app to close Teku all the time or possibly it becomes an extra param to the start methods or possibly even goes into the `TekuConfiguration`.

Reply:
Returned all `Teku` methods back to static : 3b10857
Also made static all `TekuFacade` methods. 

Shutdown hook is added just in `main` method now. Not sure if it's worth tweaking with configuration. Makes sense to leave this on the calling client 

I think it probably makes sense to refactor `BeaconNodeCommand` to return `Node` instead of `Integer` and pass all errors via exception(s). Though it could be handled in a different PR